lms/djangoapps/badges/api/views.py

```diff
@@ -0,0 +1,123 @@
+from .serializers import BadgeAssertionSerializer
+from xmodule_django.models import CourseKeyField
+
+
+class CourseKeyError(APIException):
+    """
+    Raised the course key given isn't valid.
+    """
+    status_code = 400
+    default_detail = "The course key provided could not be parsed."
+
+
+@view_auth_classes(is_user=True)
+class UserBadgeAssertions(generics.ListAPIView):
+    """
+    ** Use cases **
+
+        Request a list of assertions for a user, optionally constrained to a course.
+
+    ** Example Requests **
+
+        GET /api/badges/v1/assertions/user/{username}/
+
+    ** Response Values **
+
+        Body comprised of a list of objects with the following fields:
+
+        * user: The API Endpoint URL for the user
+        * badge_class: The badge class the assertion was awarded for. Represented as an object
+          with the following fields:
+            * slug: The identifier for the badge class
+            * issuing_component: The software component responsible for issuing this badge.
+            * display_name: The display name of the badge.
+            * course_id: The course key of the course this badge is scoped to, or null if it isn't scoped to a course.
+            * description: A description of the award and its significance.
+            * criteria: A description of what is needed to obtain this award.
+            * image_url: A URL to the icon image used to represent this award.
+        * image_url: The baked assertion image derived from the badge_class icon-- contains metadata about the award
+          in its headers.
+        * assertion_url: The URL to the OpenBadges BadgeAssertion object, for verification by compatible tools
+          and software.
+
+    ** Params **
+
+        * slug (optional): The identifier for a particular badge class to filter by.
+        * issuing_component (optional): The issuing component for a particular badge class to filter by
+          (requires slug to have been specified, or this will be ignored.) If slug is provided and this is not,
+          assumes the issuing_component should be empty.
+        * course_id (optional): Returns assertions that were awarded as part of a particular course. If slug is
+          provided, and this field is not specified, assumes that the target badge has an empty course_id field.
+          '*' may be used to get all badges with the specified slug, issuing_component combination across all courses.
+
+    ** Returns **
+
+        * 200 on success, with a list of Badge Assertion objects.
+        * 403 if a user who does not have permission to masquerade as
+          another user specifies a username other than their own.
+        * 404 if the specified user does not exist
+
+        {
+            "count": 7,
+            "previous": null,
+            "num_pages": 1,
+            "results": [
+                {
+                    "user": "http://example.com/api/mobile/v0.5/users/honor",
```

Comment:
It doesn't make sense for the user URL to link to the deprecated mobile_api.
Also, I'm not sure which URL makes sense to specify here for the user.  UserAPI?  OAuth User?
Rather, can we remove this field?  I don't think it's necessary for the client since the client already knows which user it is requesting this API for.
Otherwise, it can just be the user's username.